features/netsocket/emac-drivers/TARGET_ARM_SSG/COMPONENT_SMSC9220/smsc9220_emac.cpp

```diff
@@ -169,12 +177,12 @@ bool SMSC9220_EMAC::link_out(emac_mem_buf_t *buf)
                                       (const char*)_memory_manager->get_ptr(buf),
                                       _memory_manager->get_len(buf));
         _memory_manager->free(buf);
-        if (error != SMSC9220_ERROR_NONE) {
-            _TXLockMutex.unlock();
+        _TXLockMutex.unlock();
+        if (error == SMSC9220_ERROR_NONE) {
```

Comment:
Could be simplified to `return error == SMC9220_ERROR_NONE`.

Reply:
Simplified the return statement as suggested.